module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,58 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+void printMap(const std::array<std::array<uint8_t, width>, height>& bitmap) {
+    for (auto arr : bitmap) {
+        for (auto el : arr) {
+            if (iscntrl(el))
+                std::cout << " ";
+            else
+                std::cout << el;
+        }
+        std::cout << "\n";
+    }
+}
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& grayScale) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressed;
+    compressed.reserve(width * height);
+    uint8_t count = 0;
+
+    for (auto row : grayScale) {
+        auto _prev = row.front();
+        for (auto column : row) {
+            if (_prev == column) {
+                count++;
+                _prev = column;
+                continue;
+            }
+            compressed.push_back({_prev, count});
+            _prev = column;
+            count = 1;
+        }
+        compressed.push_back({_prev, count});
```

Comment:
you can try and simplify this to use only one one emplace_back

Reply:
Agree. I know you can do that ;)

Reply:
I tried but i think i can't xd

Reply:
Yes, you can. Try rearrange if statement:
push_back color and counter when next color is different. Otherwise increment counter.